app/src/main/java/com/kms/cura/view/activity/PatientSignUpActivity.java

```diff
@@ -76,10 +76,36 @@ public void unregisterEvent() {
 
     @Override
     public void onTextChanged(CharSequence s, int start, int before, int count) {
-        validateFirstName();
-        validateEmail();
-        validatePassword();
-        validatePasswordReenter();
+        if (edtFirstName.hasFocus()) {
+            if (!validateEmptyName()) {
+                edtFirstName.setError(getResources().getString(R.string.FirstNameError));
+            } else if (!validateFirstName()) {
+                edtFirstName.setError(getResources().getString(R.string.first_name_error));
+            } else {
+                edtFirstName.setError(null);
+            }
+        }
+        if (edtEmail.hasFocus()) {
+            if (!validateEmail()) {
+                edtEmail.setError(getResources().getString(R.string.EmailError));
+            } else {
+                edtEmail.setError(null);
+            }
+        }
+        if (edtPassword.hasFocus()) {
+            if (!validatePassword()) {
+                edtPassword.setError(getResources().getString(R.string.PasswordError));
+            } else {
+                edtPassword.setError(null);
+            }
+        }
+        if (edtPasswordReenter.hasFocus()) {
```

Comment:
all the hasFocus check should be the first line in all the validate...() method, if it is false then return. It will make the code less complicated